addons/payment/models/payment_transaction.py

```diff
@@ -926,7 +926,7 @@ def _create_payment(self, **extra_create_values):
         self.ensure_one()
 
         payment_method_line = self.acquirer_id.journal_id.inbound_payment_method_line_ids\
-            .filtered(lambda l: l.code == self.provider)
+            .filtered(lambda l: l.code == self.provider and l.payment_acquirer_id == self.acquirer_id)
```

Comment:
```suggestion
            .filtered(lambda l: l.payment_acquirer_id == self.acquirer_id)
```
I think the first condition is no longer relevant with your fix

Reply:
Sure. Done :slightly_smiling_face: